content: draft: Source VSAs shouldn't be produced for all revisions. (#1336)
slsa-framework/slsa: docs/spec/draft/source-requirements.md

<td>✓<td>✓<td>✓
<tr id="source-summary"><td>Source Summary Attestations<td>

The SCS MAY generate a [source summary attestation](#summary-attestation) to
indicate the SLSA Source Level of a revision.

> The summary attestation "interprets" what the data in the source provenance attestation means.

Reviewer: What do you mean by this? Any interpretations are also only possible if the source provenance attestation exists.

Author: I realized I hadn't actually _pushed_ the changes I made, which should address this... Sorry!

PTAL at the new lanaguge?

Reviewer: It is odd that the summary attestation is required but we state that the SCS MAY create it. I get what you are saying, it just doesn't sit right.

Should we say something like
```
The summary attestation "interprets" what the data in the source provenance attestation means if it exists for the same commit.
```

Author: Oh the MAY stuff is the old language.  If you check out the latest revision it says

"The SCS MUST generate a [source summary attestation](#summary-attestation) to
indicate the SLSA Source Level of revisions it wishes to make claims about."

and then later

"When source provenance is available the SCS MAY use it to generate the
source summary attestation."

Reviewer: Interesting. Not sure how I got stuck between revisions. I like the latest!

<td>✓<td>✓<td>✓
<tr id="source-summary"><td>Source Summary Attestations<td>

> The SCS MUST generate [source summary attestations](#summary-attestation) to
> indicate the SLSA Source Level of a given revision.

Reviewer 2: I suspect users will get a little confused between the two types of attestations. 

```suggestion
The SCS MAY generate a [source summary attestation](#summary-attestation) to
indicate the SLSA Source Level of a revision.
The summary attestation "interprets" what the data in the source provenance attestation means.
```

If a consumer is authorized to access a revision, they MUST be able to fetch the
corresponding source summary attestations.

> It is possible that an SCS makes no claims about a particular revision.

Reviewer 2: it seems like line 176 is saying they must, but here they don't have to. 
I think it's that they MAY.

> It is possible that an SCS makes no claims about a particular revision.
> In these cases the SCS MAY NOT generate a summary attestation. Consumers MUST
> interpret the absence of a source summary attestation as though the revision
> is SLSA Source Level 0.

Reviewer 2: ```suggestion
If the SCS DOES NOT generate a summary attestation for a revision, the revision MUST be verified as having SLSA Source Level 0.
```

or maybe "cannot be verified and thus has source lvl 0?"

Author: I'll take your suggestion and tweak, thanks.

<td>✓<td>✓<td>✓
<tr id="source-summary"><td>Source Summary Attestations<td>

> The SCS MUST generate a [source summary attestation](#summary-attestation) to
> indicate the SLSA Source Level of revisions it wishes to make claims about.

Reviewer 2: ```suggestion
The SCS MAY generate a [source summary attestation](#summary-attestation) to
indicate the SLSA Source Level of a revision.
```
Does "must if you wish" mean "may"?

Author: "If you want to claim a revision has a SLSA level you MUST create a VSA for it"

Author: Removed the wishes bit.  I think this covers it. :)
